src/services/openid_connect.cpp

```diff
@@ -0,0 +1,249 @@
+#include "portal/basketapi.h"
+#include "services/httpservice.h"
+#include "userdb/userdb.h"
+#include "util/concat.h"
+#include "util/configuration.h"
+#include "util/curl.h"
+#include "util/exceptions.h"
+
+#include <cstring>
+#include <json/json.h>
+#include <algorithm>
+#include <pqxx/pqxx>
+#include <jwt/jwt.hpp>
+#include <cppcodec/base64_url_unpadded.hpp>
+
+/// This class provides methods for user authentication with OpenId Connect
+class OpenIdConnectService : public HTTPService {
+    public:
+        using HTTPService::HTTPService;
+
+        ~OpenIdConnectService() override = default;
+
+        struct OpenIdConnectServiceException
+                : public std::runtime_error {
+            using std::runtime_error::runtime_error;
+        };
+
+    protected:
+        /// Dispatch requests
+        auto run() -> void override;
+
+        /// Login using an OpenId Connect access token
+        auto login(const std::string &access_token) const -> void;
+
+    private:
+        struct User {
+            std::string goe_id; // identifier
```

Comment:
please add some explanation what this goe_id does

Reply:
There is a comment that this is the identifier - so what additional information do you need?

Reply:
maybe just add that its the id of the goettingen sso provider